bin/garden-test

```diff
@@ -0,0 +1,280 @@
+                echo "test ${test} is enabled by the following features: ${enabledBy}"
+                # prepare all other dependencies from the test
+                for fd in $(echo "${enabledBy}"); do
+                        if [ -d "${featureDir}/${fd}/test/${test}.d" ]; then
+                                mkdir -p "${rootfs}/tmp/${test}.d"
+                                for fdep in "${featureDir}/${fd}/test/${test}.d"/*; do
+                                        [ -e "$fdep" ] || continue
+                                        fdepshort=$(basename $fdep)
+                                        if [[ "$fdepshort" == "pkg.include" || "$fdepshort" == "pkg.exclude" ]]; then
+                                                cat $fdep | filter_variables | filter_if >> "${rootfs}/tmp/${test}.d/${fdepshort}.list"
+                                        else
+                                                cat $fdep >> "${rootfs}/tmp/${test}.d/${fdepshort}"
+                                        fi
+                                done
+                        fi
+                done
+
+                # move the actual tests from one of the features that enables it
+                actualTest="${featureDir}/$(echo ${enabledBy} | awk '{ print $1 }')/test/${t}"
+
+		# Test: chroot 
+                if [ ${t##*.} == "chroot" ]; then
+                        cp -L ${actualTest} "${rootfs}/tmp/${test}"
+                        if garden-chroot "${rootfs}" ./tmp/${test}; then
+                                echo -e "\e[32mpassed\e[39m"
+                                echo
+                        else
+                                echo -e "\e[31mfailed\e[39m"
+                                echo
+                                let "failcounter=failcounter+1"
+                        fi
+
+		# Test: startvm (running in qemu)
+                elif [ ${t##*.} == "startvm" ]; then
+                        cp -L ${actualTest} "${rootfs}/tmp/${test}"
+			# Strip trailing whitespace
+			startvmUser=$(echo ${enabledBy} | head -c-1)
+			# Create a temporary and only for unit tests valid
+			# ssh key for each user and feature
+			garden-chroot "${rootfs}" useradd -m ${startvmUser}
+			garden-chroot "${rootfs}" adduser ${startvmUser} wheel
+			garden-chroot "${rootfs}" mkdir -p /home/${startvmUser}/tests/
+			garden-chroot "${rootfs}" mkdir -p /home/${startvmUser}/.ssh/
+			garden-chroot "${rootfs}" chmod 700 /home/${startvmUser}/.ssh/
+			/usr/bin/ssh-keygen -q -t rsa -N '' -f /rootfs/home/${startvmUser}/.ssh/id_rsa -m pem
+			cp /rootfs/home/${startvmUser}/.ssh/id_rsa.pub /rootfs/home/${startvmUser}/.ssh/authorized_keys
+			garden-chroot "${rootfs}" chmod 600 /home/${startvmUser}/.ssh/authorized_keys
+			garden-chroot "${rootfs}" chown -R ${startvmUser}:${startvmUser} /home/${startvmUser}/.ssh/
+			# Copy whole test directory to features home directory
+			rsync -aP "${featureDir}/${startvmUser}/test/" /rootfs/home/${startvmUser}/tests/
```

Comment:
Why do we copy the tests into the rootfs again, if they are already copied into the rootfs some lines above?
The tests will be underneath the `/tmp/<test>/<test>.d` directory already.

Reply:
Yep, it's */tmp* and the content isn't present anymore when the VM boots up in QEMU. Maybe we should move the overall tests to a new persistent directory since the new test image is dropped in any case. But this should be done in a new PR.